Lib/uuid.py

```diff
@@ -388,6 +392,48 @@ def _find_mac(command, args, hw_identifiers, get_index):
         pass
     return first_local_mac or None
 
+def _find_mac_netstat(command, args, hw_identifiers, get_index):
```

Comment:
I don't understand why this function is needed? Why not leaving the code in _netstat_getnode()? I don't understand the purpose of get_index. It seems like your design is to be able to write more easily tests. But you can mock a lock of things, especially the uuid._AIX constant. You can test the AIX-specific code on all platforms instead of skipping the test on other platforms. Otherwise, if the AIX code is broken, nobody will notice it until someone runs the tests on AIX.

Reply:
a) not to be creative - more because you asked for an AIX specific test but also to duplicate/copy the way _find_mac() is used by the other routines - as well as tested by test_uuid.
b) _find_mac is targeting output that has "keyword value" pairs on the same line, while _find_mac_netstat (maybe a different name is needed) is for output where the first line gives the position of values to consider. In short, _find_mac cannot parse 'netstat' output, so rather than have the code twice I thought it prudent to move the logic outside of _netstat_getnode().
c) as to being able to mock in other ways - no doubt. But this is what I could come up with the skill level I have. Suggestions for a better way are welcome. I like to learn - and even doing this helped my learn a lot.
d) the code is not checked via the fake data (if not AIX), but it is checked via the other test(s) that call _netstat_getnode(). However, if additional fake data is provided for other platforms the the test can also be used for them. This is what I can provide - and test manually.